### DISCUSSION TITLE
ImportError: cannot import name '_cext'

### DISCUSSION THREAD

**Discussion by mrkhensu** [shap/shap#2536] [Q&A]
Hello there,
I am getting the following problem for the program that I will share the sample codes of; "ImportError: cannot import name '_cext'".
please help me.

```
import shap
explainer = shap.TreeExplainer(model)
shap_values = explainer.shap_values(X_train)

shap.initjs()
shap.force_plot(explainer.expected_value, shap_values[0,:], X_train.iloc[0,:])
```
PROBLEM:

File "C:\Users\MEHMET\.spyder-py3\shap\explainers\_tree.py", line 24, in <module>
    from .. import _cext

ImportError: cannot import name '_cext' from partially initialized module 'shap' (most likely due to a circular import) (C:\Users\MEHMET\.spyder-py3\shap\__init__.py)

**Comment by Linda-L** [comment]
me too

**Comment by Linda-L** [comment]
解决了，是因为有有循环套用，
![image](https://github.com/user-attachments/assets/ae56ccbc-825d-48ab-9398-79ada3ecc74c)
我把例子文件放到新开的一个pycharm项目中就好了
